src/frontend/js/components/PurchaseButton/index.tsx

```diff
@@ -33,21 +35,26 @@ const messages = defineMessages({
 
 interface PurchaseButtonProps {
   product: Joanie.Product;
-  disabled: boolean;
+  disabled?: boolean;
+  className?: string;
 }
 
-const PurchaseButton = ({ product, disabled }: PurchaseButtonProps) => {
+const PurchaseButton = ({ product, disabled = false, className }: PurchaseButtonProps) => {
   const intl = useIntl();
   const { user, login } = useSession();
   const [isSaleTunnelOpen, setIsSaleTunnelOpen] = useState(false);
-
   const isOpenedCourseRun = (courseRun: Joanie.CourseRun) =>
     courseRun.state.priority <= Priority.FUTURE_NOT_YET_OPEN;
 
   const hasAtLeastOneCourseRun = useMemo(() => {
+    // FIXME(rlecellier): define some constraint messages like:
+    // - course run in the past: false
+    if (product.type === Joanie.ProductType.CERTIFICATE) {
+      return true;
+    }
     return (
       product.target_courses.length > 0 &&
-      !product.target_courses.some(({ course_runs }) => !course_runs.some(isOpenedCourseRun))
+      product.target_courses.every(({ course_runs }) => course_runs.some(isOpenedCourseRun))
     );
   }, [product]);
```

Comment:
This logic is tied to CREDENTIAL products. I wonder if we do not move this logic upper and just pass a prop disable to the `PurchaseButton` 🤔